src/main/daml/Daml/Finance/Instrument/Swap/Fpml/Util.daml

```diff
@@ -327,7 +329,114 @@ roundRate rate rounding =
         Down -> RoundingFloor
         Nearest -> RoundingHalfUp
 
--- | Create claims from swapStream that describes a fixed or floating coupon stream.
+-- | Create claims from swapStream that describes a floating coupon stream.
+-- This function can handle compounding of rates (currently only straight compounding)
+-- Integrate this in calculateFloatingPaymentClaimsFromSwapStream once the other compounding
+-- methods have been implemented.
+calculateCompoundedFloatingPaymentClaimsFromSwapStream : FloatingRateCalculation -> SwapStream ->
+  PeriodicSchedule -> [SchedulePeriod] -> [SchedulePeriod] -> Bool -> Bool -> Deliverable ->
+  Party -> Party -> Optional Text -> Optional [Date] -> [(Decimal, Bool)] -> Update [TaggedClaim]
+calculateCompoundedFloatingPaymentClaimsFromSwapStream floatingRateCalculation s periodicSchedule
+  calculationSchedule paymentSchedule useAdjustedDatesForDcf issuerPaysLeg currency issuer
+  calendarDataAgency fxRateId fxFixingDates notionals = do
```

Comment:
For multiline function declarations, match up the types with their respective variable name definitions to simplify for readers :

```suggestion
calculateCompoundedFloatingPaymentClaimsFromSwapStream : FloatingRateCalculation -> SwapStream ->
  PeriodicSchedule -> [SchedulePeriod] -> [SchedulePeriod] -> Bool -> Bool -> Deliverable ->
  Party -> Party -> Optional Text -> Optional [Date] -> [(Decimal, Bool)] -> Update [TaggedClaim]
calculateCompoundedFloatingPaymentClaimsFromSwapStream floatingRateCalculation s 
periodicSchedule calculationSchedule paymentSchedule useAdjustedDatesForDcf issuerPaysLeg currency 
issuer calendarDataAgency fxRateId fxFixingDates notionals = do
```